### DISCUSSION TITLE
[Question] Can I switch seamlessly between Keycloak 15.1.1 and Keycloak 18.0.0?

### DISCUSSION THREAD

**Discussion by Paul6552** [keycloak/keycloak#12077] [Q&A]
### Describe the bug

I see a lot of people here having reverse proxy nginx and keycloak 18 issues.

I also wanted to switch from version 15.1.1 to 18.0.0, but only in parallel operation to test the changeover well.

I'm just afraid that once I move to version 18 I won't be able to go back to 15.1.1.
If I leave keycloak 18 on the production database, does keycloak 18 update any database entries?

Is it possible to seamlessly switch back and forth between 15.1.1 and 18?

### Version

15.1.1

**Comment by darius-m** [comment]
At the least, the Keycloak version number stored in the database is changed, but with new features in each version, the database structure is bound to differ at least a bit. You should back up your database before attempting an upgrade, so you can restore it easily if the upgrade fails. You can find some of the changes that appeared in the newer versions in the [upgrade guide](https://www.keycloak.org/docs/latest/upgrading/index.html#migration-changes).

As a note, I tried to upgrade from 15.1.1 to 18.0.0 legacy and failed because the parameters required to connect to the infinispan service seem to change when the config is read from the file. This may not be an issue unless you have a multi-server deployment.
